targets/esp8266/jshardware.c

```diff
@@ -588,15 +593,47 @@ void jshUSARTKick(
 //===== SPI =====
 
 /**
- * Unknown
+ * Initialize the hardware SPI device.
+ * On the ESP8266, hardware SPI is implemented via a set of pins defined
+ * as follows:
  *
+ * | GPIO   | NodeMCU | Name  | Function |
+ * |--------|---------|-------|----------|
+ * | GPIO12 | D6      | HMISO | MISO     |
+ * | GPIO13 | D7      | HMOSI | MOSI     |
+ * | GPIO14 | D5      | HSCLK | CLK      |
+ * | GPIO15 | D8      | HCS   | CS       |
  *
  */
 void jshSPISetup(
-    IOEventFlags device, //!< Unknown
-    JshSPIInfo *inf      //!< Unknown
+    IOEventFlags device, //!< The identity of the SPI device being initialized.
+    JshSPIInfo *inf      //!< Flags for the SPI device.
   ) {
-  os_printf("ESP8266: jshSPISetup: device=%d, inf=0x%x\n", device, (int)inf);
+  // The device should be one of EV_SPI1, EV_SPI2 or EV_SPI3.
+  os_printf("> jshSPISetup - jshSPISetup: device=%d\n", device);
+  switch(device) {
+  case EV_SPI1:
+    os_printf(" - Device is SPI1\n");
+    // EV_SPI1 is the ESP8266 hardware SPI ...
+    spi_init(HSPI); // Initialize the hardware SPI components.
+    spi_clock(HSPI, CPU_CLK_FREQ / (inf->baudRate * 2), 2);
+    g_spiInitialized = true;
+    break;
+  case EV_SPI2:
+    os_printf(" - Device is SPI2\n");
+    break;
+  case EV_SPI3:
+    os_printf(" - Device is SPI3\n");
+    break;
+  default:
+    os_printf(" - Device is Unknown!!\n");
+    break;
+  }
+  if (inf != NULL) {
+    os_printf("baudRate=%d, baudRateSpec=%d, pinSCK=%d, pinMISO=%d, pinMOSI=%d, spiMode=%d, spiMSB=%d\n",
+        inf->baudRate, inf->baudRateSpec, inf->pinSCK, inf->pinMISO, inf->pinMOSI, inf->spiMode, inf->spiMSB);
```

Comment:
These guards are an over-abundance of caution.  The identies of available SPI interfaces are not user configurable but instead defined in the ESP8266_xxx.py files.  Today we only have SPI1 but it is possible to have SPI2 and SPI3 as well.  However, nothing a user can code at their JS level could result in triggering these alternate code paths.